Gems/AtomTressFX/Assets/Shaders/HairLightingEquations.azsli

```diff
@@ -0,0 +1,244 @@
+    fresnel = Pow2(1 - fresnel) * fresnel;
+
+    // How much light the hair will absorb. Part of the attenuation term (A in the papers)
+    float3 absorption = pow(surface.albedo, 0.8f / max(cos_Ld, 0.001) );
+
+    // Distribution term
+    float distribution = exp(17.0f * cos_O - 16.78f);
+
+    return absorption * (fresnel * distribution);
+}
+
+//------------------------------------------------------------------------------
+//                    The BSDF lighting function used by Hair
+//------------------------------------------------------------------------------
+float3 HairMarschnerBSDF(Surface surface, LightingData lightingData, const float3 dirToLight)
+{
+    //-------------- Lighting Parameters Calculations ---------------
+    // Incoming and outgoing light directions
+    float3 Wi = normalize(dirToLight);                 // Incident light direction
+    float3 Wr = normalize(lightingData.dirToCamera);   // Reflected light measurement direction AKA reflection                                        
+    float3 T = normalize(surface.tangent);             // Hair tangent
+
+    // Incident light and reflection direction projected along the tangent
+    float Ti = T * dot(T, Wi);
+    float Tr = T * dot(T, Wr);
+
+    // The light and reflection vectors projected along the normal plane to the hair.
+    // The plane is spliting the Azimuth and Longtitude angles and vectors contributions.
+    float3 NPi = normalize(Wi - Ti);
+    float3 NPr = normalize(Wr - Tr);
+
+    // Azimuthal angle between the incident light vector and the reflection 
+    // direction (the direction at which we measure the light scaterring)
+    // float O = acos(dot(NPi, NPr))  <- Unused, for reference only
+    float cos_O = dot(NPi, NPr);
+
+    // cosine(O / 2)
+    float cos_O2 = sqrt(0.5 * cos_O + 0.5);  // <- trigonometric formula for calculating cos(x/2) given cos(x) 
+
+    // Longitude angles
+    float Li = acos(clamp(dot(Wi, NPi),-1.0f, 1.0f));
+    float Lr = acos(clamp(dot(Wr, NPr),-1.0f, 1.0f));
+    float Lh = (Lr + Li) * 0.5f;
+    float Ld = (Lr - Li) * 0.5f;
+
+    // The folowing is according to the original article - reference only
+    //    float sinLiPlusSinLr = dot(Wi, NPi) * dot(Wr, NPr);// sin(Li) + sin(Lr);
+    float sinLiPlusSinLr = sin(Li) + sin(Lr);
+
+    // Refraction index 
+    const float n = 1.55f;
+    float cos_Ld = cos(Ld);
+    float n2 = (1.19f / cos_Ld) + 0.36f * cos_Ld;
+
+    // Fresnel F0
+    float f0 = Pow2( (1.0f - n) / (1.0f + n) );
+
+    //--------------- Lighting accumulation per lobe ------------------
+    float3 lighting = float3(0.0f, 0.0f, 0.0f);
+
+    // R Path - single reflection from the hair towards the eye.
+    if (o_enableMarschner_R)
+    {
+        float lighting_R = o_enableDiffuseLobe ? M_R(surface, Lh, sinLiPlusSinLr) : float3(1.0f, 1.0f, 1.0f);
+        if (o_enableSpecularLobe)
+            lighting_R *= N_R(surface, cos_O2, Wi, Wr, f0);
```

Comment:
o_enableSpecularLobe is a bit of a weird name, since both M_R and N_R are contributing to specular lighting.

Reply:
Absolutely - this started as debug aid without knowing what would be required.
It should be LongitudeCoefficient and AzimuthCoefficient.
I will leave this remark live until changed and tested - tnx.